packages/playground/src/components/Playground.tsx

```diff
@@ -0,0 +1,357 @@
+      location: { origin, pathname },
+    } = document;
+    try {
+      const hash = btoa(
+        JSON.stringify({
+          formData,
+          schema,
+          uiSchema,
+          theme,
+          liveSettings,
+        })
+      );
+
+      setShareURL(`${origin}${pathname}#${hash}`);
+    } catch (err) {
+      setShareURL(null);
+    }
+  };
+
+  let templateProps: any = {};
+  if (ArrayFieldTemplate) {
+    templateProps.ArrayFieldTemplate = ArrayFieldTemplate;
+  }
+  if (ObjectFieldTemplate) {
+    templateProps.ObjectFieldTemplate = ObjectFieldTemplate;
+  }
+  if (extraErrors) {
+    templateProps.extraErrors = extraErrors;
+  }
+
+  return (
+    <>
+      <div className='page-header'>
+        <h1>react-jsonschema-form</h1>
+        <div className='row'>
+          <div className='col-sm-6'>
+            <Selector onSelected={load} />
+          </div>
+          <div className='col-sm-2'>
+            <Form
+              idPrefix='rjsf_options'
+              schema={liveSettingsSchema as any}
+              formData={liveSettings}
+              validator={localValidator}
+              onChange={handleSetLiveSettings as any}
+            >
+              <div />
+            </Form>
+          </div>
+          <div className='col-sm-2'>
+            <ThemeSelector themes={themes} theme={theme} select={onThemeSelected} />
+            {themes[theme] && themes[theme].subthemes && (
+              <SubthemeSelector subthemes={themes[theme].subthemes} subtheme={subtheme} select={onSubthemeSelected} />
+            )}
+            <ValidatorSelector validators={validators} validator={validator} select={onValidatorSelected} />
+            <button
+              title='Click me to submit the form programmatically.'
+              className='btn btn-default'
+              type='button'
+              onClick={() => playGroundFormRef.current.submit()}
+            >
+              Prog. Submit
+            </button>
+            <span> </span>
+            <button
+              title='Click me to validate the form programmatically.'
+              className='btn btn-default'
+              type='button'
+              onClick={() => {
+                const valid = playGroundFormRef.current.validateForm();
+                alert(valid ? 'Form is valid' : 'Form has errors');
+              }}
+            >
+              Prog. Validate
+            </button>
+            <div style={{ marginTop: '5px' }} />
+            <CopyLink shareURL={shareURL} onShare={onShare} />
+          </div>
+          <div className='col-sm-2'>
+            <RawValidatorTest validator={validators[validator]} schema={schema} formData={formData} />
+          </div>
+        </div>
+      </div>
+      <Editors
+        {...{
+          extraErrors,
+          formData,
+          schema,
+          setExtraErrors,
+          setFormData,
+          setSchema,
+          setShareURL,
+          setUiSchema,
+          uiSchema,
+        }}
```

Comment:
Why not be more explicit?
```suggestion
        extraErrors={extraErrors},
        formData={formData}
        schema={schema}
        setExtraErrors={setExtraErrors}
        setFormData={setFormData}
        setSchema={setSchema}
        setShareURL={setShareURL}
        setUiSchema={setUiSchema}
        uiSchema={uiSchema}
```